<?xml version="1.0"?>
<robot name="SMALdog">

  <material name="Orange">
    <color rgba="1.0  0.55 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>

  <link name="body_link">
    <visual>
      <origin xyz="0.0 0.0 -0.0155" rpy="0 0 0"/>
      <geometry>
        <box size="0.2 0.065 0.055"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>

  <joint name="lf_pitch_joint" type="revolute">
    <axis xyz="1 0 0" />
    <origin xyz="0.086 0.019 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="body_link"/>
    <child link="lf_shoulder_link"/>
  </joint>
  <link name="lf_shoulder_link">
    <visual>
      <origin xyz="0.0 0.025 0" rpy="1.57 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.050"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <joint name="lf_flex_joint" type="revolute">
    <axis xyz="0 1 0" />
    <origin xyz="0 0.050 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="lf_shoulder_link"/>
    <child link="lf_femur_link"/>
  </joint>
  <link name="lf_femur_link">
    <visual>
      <origin xyz="0.0 0.0 -0.032" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.064"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <joint name="lf_knee_joint" type="revolute">
    <axis xyz="0 -1 0" />
    <origin xyz="0 0 -0.064" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="lf_femur_link"/>
    <child link="lf_tibia_link"/>
  </joint>
  <link name="lf_tibia_link">
    <visual>
      <origin xyz="0.0 0.0 -0.043" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.086"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>

  <joint name="lr_pitch_joint" type="revolute">
    <axis xyz="1 0 0" />
    <origin xyz="-0.086 0.019 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="body_link"/>
    <child link="lr_shoulder_link"/>
  </joint>
  <link name="lr_shoulder_link">
    <visual>
      <origin xyz="0.0 0.025 0" rpy="1.57 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.050"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <joint name="lr_flex_joint" type="revolute">
    <axis xyz="0 -1 0" />
    <origin xyz="0 0.050 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="lr_shoulder_link"/>
    <child link="lr_femur_link"/>
  </joint>
  <link name="lr_femur_link">
    <visual>
      <origin xyz="0.0 0.0 -0.032" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.064"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <joint name="lr_knee_joint" type="revolute">
    <axis xyz="0 1 0" />
    <origin xyz="0 0 -0.064" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="lr_femur_link"/>
    <child link="lr_tibia_link"/>
  </joint>
  <link name="lr_tibia_link">
    <visual>
      <origin xyz="0.0 0.0 -0.043" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.086"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>

  <joint name="rf_pitch_joint" type="revolute">
    <axis xyz="-1 0 0" />
    <origin xyz="0.086 -0.019 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="body_link"/>
    <child link="rf_shoulder_link"/>
  </joint>
  <link name="rf_shoulder_link">
    <visual>
      <origin xyz="0.0 -0.025 0" rpy="-1.57 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.050"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <joint name="rf_flex_joint" type="revolute">
    <axis xyz="0 1 0" />
    <origin xyz="0 -0.050 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="rf_shoulder_link"/>
    <child link="rf_femur_link"/>
  </joint>
  <link name="rf_femur_link">
    <visual>
      <origin xyz="0.0 0.0 -0.032" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.064"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <joint name="rf_knee_joint" type="revolute">
    <axis xyz="0 -1 0" />
    <origin xyz="0 0 -0.064" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="rf_femur_link"/>
    <child link="rf_tibia_link"/>
  </joint>
  <link name="rf_tibia_link">
    <visual>
      <origin xyz="0.0 0.0 -0.043" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.086"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>

  <joint name="rr_pitch_joint" type="revolute">
    <axis xyz="-1 0 0" />
    <origin xyz="-0.086 -0.019 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="body_link"/>
    <child link="rr_shoulder_link"/>
  </joint>
  <link name="rr_shoulder_link">
    <visual>
      <origin xyz="0.0 -0.025 0" rpy="-1.57 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.050"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <joint name="rr_flex_joint" type="revolute">
    <axis xyz="0 -1 0" />
    <origin xyz="0 -0.050 0" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="rr_shoulder_link"/>
    <child link="rr_femur_link"/>
  </joint>
  <link name="rr_femur_link">
    <visual>
      <origin xyz="0.0 0.0 -0.032" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.064"/>
      </geometry>
      <material name="Orange"/>
    </visual>
  </link>
  <joint name="rr_knee_joint" type="revolute">
    <axis xyz="0 1 0" />
    <origin xyz="0 0 -0.064" rpy="0 0 0"/>
    <limit effort="30" lower="-1.57" upper="1.57" velocity="0.5"/>
    <parent link="rr_femur_link"/>
    <child link="rr_tibia_link"/>
  </joint>
  <link name="rr_tibia_link">
    <visual>
      <origin xyz="0.0 0.0 -0.043" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.00635" length="0.086"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>

</robot>
